### PR TITLE
Expose renderer directly to allow non-Context JSON object values.

### PR DESCRIPTION
**Pull request by SergioBenitez**
It would be nice to render JSON values directly without first converting into a `Context` object. This is the way other Rust templating libraries work, and I think it makes the most sense. One way to do this is to export the Renderer type so that someone can call `Renderer::new_with_json` and then `render` directly. This is what this PR contributes.

### CODE REVIEW

**Comment by Keats** [comment]
I'm not convinced about the `Renderer::new_with_json` as that allows invalid calls, `render_with_json(..,..,to_value(&1)` would be valid. An alternative would be a `render` method that takes `BTreeMap<String, Value>` as arg but i'm not sure it's valuable at all.

Another thing against the `render_with_json` is that json is a temporary format for Tera, hopefully we will be able to move to something else (#45)

Also if it exists it should be on the `Tera` instance like the `render`

**Comment by SergioBenitez** [comment]
Restricting rendering to `Context` (or any non-polymorphic type, to be clear) is a considerable usability problem. Imagine a library that wants to expose an interface to Tera. With the current design, and the one you've proposed, the library is shoehorned into making one of two choices: 1) it must copy everything into a `Context`, or 2) it has to re-export `Context` and force their users into using that type.

What's the issue with allowing something that's not a `Value::Object` to be passed in? If that happens, we should simply return an error to the user. That is, `new_with_json ` should return a `Result` object if the Value isn't an object. I didn't do that here because later code (currently, though you've removed it in a PR) checks if the value is indeed a `Value::Object`.

Alternatively, you can define your own serializer and deserializer for Serde and ship that along with Tera. But this is functionally identical to what I've proposed, except it moves the `Result` up a level to the serialization step.

**Comment by Keats** [comment]
The last paragraph is the optimal path as it allows more information than what JSON offers (dates etc) but it is not an immediate goal.

You can change the type of the `Renderer` constructor to take `context` as a value and call `to_json` in `Tera::render`. This way `Tera::render_with_json` is easier as it only needs to check if the `Value` is of type `Value::Object` and then only calls the renderer with correct input.

**Comment by SergioBenitez** [comment]
Sounds good to me. Added `Tera::render_value`, which I thought sounded a bit better than `Tera::render_with_json`.

**Comment by SergioBenitez** [comment]
Actually, that sounds like it's rendering the value itself, so I changed it to `value_render`.

File: src/tera.rs [hunk 1]
```diff
@@ -57,13 +59,28 @@ impl Tera {
...
     pub fn render(&self, template_name: &str, data: Context) -> TeraResult<String> {
         let template = try!(self.get_template(template_name));
-        let mut renderer = Renderer::new(template, self, data);
+        let mut renderer = Renderer::new(template, self, data.as_json());
 
         renderer.render()
     }
 
+    /// Renders a Tera template given a `Serializeable` object.
+    pub fn value_render<T>(&self, template_name: &str, data: &T) -> TeraResult<String>
```
Comment (Keats) [inline src/tera.rs]: `render_from_value`?

**Comment by Keats** [comment]
Only missing integrations tests for render failures and render ok  for the new `render` method + some docs on README and it's ready to go

**Comment by Keats** [comment]
I'll do it, thanks!

**Pull request merged**